ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/SubmissionValidator.scala

```diff
@@ -46,14 +48,23 @@ class SubmissionValidator[LogResult](
         Map[DamlStateKey, Option[DamlStateValue]],
     ) => LogEntryAndState,
     allocateLogEntryId: () => DamlLogEntryId,
-    checkForMissingInputs: Boolean = false,
+    checkForMissingInputs: Boolean,
+    maximumStateValueCacheSize: Long,
     metricRegistry: MetricRegistry,
 )(implicit executionContext: ExecutionContext) {
 
   private val logger = ContextualizedLogger.get(getClass)
 
   private val timedLedgerStateAccess = new TimedLedgerStateAccess(ledgerStateAccess)
 
+  private val stateValueCache = CacheBuilder
```

Comment:
Would it be possible to inject this cache instance, please? I understand that with a maximum weight there will be no caching effectively. Also, we may want to have more sophisticated caching logic applied such as one only caching packages, etc.